addons/website/views/website_templates.xml

```diff
@@ -173,7 +173,7 @@
         <!-- website.is_public_user() is needed for menus having a page with
         restricted visibility (only shown to logged in user): public users and
         logged in users can't share the menu cache. -->
-        <attribute name="t-cache">xmlid,website,website.is_public_user()</attribute>
+        <attribute name="t-cache">xmlid,website,website.is_public_user(),main_object._name != 'website.page' and main_object</attribute>
```

Comment:
So that's because both record pages are sharing the same xmlid despite being different record, thus sharing the same cache key :grimacing: ?

This is too violent imo, the cache is going to explode, eg we have 60.000 visible forum posts, each will now have their own menu cache instead of sharing one

We will have to find another way, disabling the cache in the case of a record url set in menu would even be better, but we can't know that easily..

At the end, I am just wondering why aren't we doing that in js instead of python, it's like that since forever but maybe not the best approach

Reply:
We did that in python and it was a good idea and then the t-cache appears and make it complicated to handle correctly... I will do a public widget that set the active class on the correct navbar elements. It will be better but there will be a "lag" between the draw of the HTML of the navbar and the JS code that will add the active class on elements... 

Maybe the best from the UX point of view would be to mix both solutions. Keep the same system as we have now but if there is a record related link in the navbar the JS handle it. It could work but not a good thing from technical POV. It's best to always let the JS handle it. :+1:

Reply:
I think it's ok to have both, we first try to ship the DOM already with correct "active", and worst case the JS corrects it in the rare case like this one.